### ISSUE TITLE
Missing override

### ISSUE THREAD

**Issue by heddendorp** [dotansimha/graphql-code-generator#7478]
https://github.com/dotansimha/graphql-code-generator/blob/aeaea1ed1d09172a3bf8a9bd4f1d3cd7ab65023b/packages/plugins/typescript/apollo-angular/src/visitor.ts#L302

Should be fixed to prevent `error TS4114: This member must have an 'override' modifier because it overrides a member in the base class`

**Comment by RyannGalea** [comment]
Yes, I'm having the same issue where document needs 'override document'

**Comment by TomONeill** [comment]
See `addExplicitOverride` on https://www.graphql-code-generator.com/plugins/typescript-apollo-angular

**Comment by TomONeill** [comment]
This was fixed with release [@graphql-codegen/typescript-apollo-angular@3.4.0](https://github.com/dotansimha/graphql-code-generator/releases/tag/%40graphql-codegen%2Ftypescript-apollo-angular%403.4.0)

Add `addExplicitOverride: true` to your config.
See https://www.graphql-code-generator.com/plugins/typescript-apollo-angular.

**Comment by inpercima** [comment]
@TomONeill Thank you for your answer. This works.

**Comment by yannickglt** [comment]
Hi, 
Thanks to your answer but it does not sound to work with the `client` property if overridden `namedClient`.
Do you have an alternative for this one?
Thanks in advance.

**Comment by TomONeill** [comment]
I added an issue for this (#7553) and a PR (#7548), but so far it hasn't been approved yet.

**Comment by yannickglt** [comment]
Amazing! Thanks a lot. Will disable the check in the tsconfig for now 😊

**Comment by TomONeill** [comment]
PR was merged! I think you can close the issue now.

**Comment by CemYil03** [comment]
I still have this issue with "@graphql-codegen/typescript-apollo-angular": "^3.4.12" installed :(

**Comment by Gkleinereva** [comment]
If all else fails, you should be able to set the "noImplicitOverride" compiler option to false in your tsconfig.json file.  Depending on how you generated your application, this may have been set to true for you by default.

**Comment by wi4er** [comment]
"@graphql-codegen/typescript-apollo-angular": "^3.5.5" Problem still there.

**Comment by akialapiha** [comment]
> 

Ran into this issue too (with version 3.5.5) but as stated before, the problem no longer exists. All I had to do was to set the `addExplicitOverride` config flag to `true` as stated in the [documentation](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-apollo-angular).



_codegen.ts_
```
import type { CodegenConfig } from '@graphql-codegen/cli';
 
 const config: CodegenConfig = {
   // ...
   generates: {
     'path/to/file': {
       plugins: ['apollo-angular'],
       config: {
         addExplicitOverride: true
       },
     },
   },
 };
 export default config;
 ``` 

The issue should be closed.
